src/Microsoft.ML.Data/Utilities/ColumnCursor.cs

```diff
@@ -20,16 +20,21 @@ public static class ColumnCursorExtensions
         /// </summary>
         /// <typeparam name="T">The type of the values. This must match the actual column type.</typeparam>
         /// <param name="data">The data view to get the column from.</param>
-        /// <param name="env">The current host environment.</param>
-        /// <param name="columnName">The name of the column to extract.</param>
-        public static IEnumerable<T> GetColumn<T>(this IDataView data, IHostEnvironment env, string columnName)
+        /// <param name="column">The column to be extracted.</param>
+        public static IEnumerable<T> GetColumn<T>(this IDataView data, DataViewSchema.Column column)
```

Comment:
>column [](start = 93, length = 6)

I would prefer to have string version of this method as well, but I have impression Tom doesn't like that idea (not sure why).
No blocker, just fishing for explanation.

Reply:
Maybe for consistency with static APIs?

Reply:
> I would prefer to have string version of this method as well, but I have impression Tom doesn't like that idea (not sure why).

I think this is how religious doctrine can get established in some cases. :)

But seriously, I think the *core* interfaces of `IDataView` itself and its accessors ought to work over `Schema.Column`. It's just safer. A convenience *like this* however, surely benefits from a `string`. I think it's quite fine to have both; in the one case, you get some degree of "safety." In the other, convenience. This is clearly a case where convenience *is the intent*, and insisting on a column is inconvenient.

---
In reply to: [260585414](https://github.com/dotnet/machinelearning/pull/2738#discussion_r260585414) [](ancestors = 260585414)

Reply:
String overloading is added in Iteration 7.

---
In reply to: [260860073](https://github.com/dotnet/machinelearning/pull/2738#discussion_r260860073) [](ancestors = 260860073,260585414)